src/scales/scale.linear.js

```diff
@@ -139,11 +139,11 @@ module.exports = function(Chart) {
 			var tickOpts = me.options.ticks;
 
 			if (me.isHorizontal()) {
-				maxTicks = Math.min(tickOpts.maxTicksLimit ? tickOpts.maxTicksLimit : 11, Math.ceil(me.width / 50));
+				maxTicks = Math.min(tickOpts.maxTicksLimit ? tickOpts.maxTicksLimit : 11, Math.ceil(me.width / 40));
```

Comment:
`11` and `40` are arbitrary values, right? maybe we should replace `11` by something computed based on `stepSize`?

Reply:
I'm thinking of the following improvement:
```js
if (stepSize > 0) {
    maxTicks = Math.ceil(me.max / stepSize) - Math.floor(me.min / stepSize) + 1;
} else if (me.isHorizontal()) {
    maxTicks = Math.ceil(me.width / 40);
} else {
    var tickFont = helpers.options._parseFont(tickOpts);
    maxTicks = Math.ceil(me.height / tickFont.lineHeight);
}
if (maxTicksLimit || !(stepSize > 0)) {
    maxTicks = Math.min(maxTicksLimit || 11, maxTicks);
}
```
I think having `11` as a limit is still useful in case that `stepSize` is not set.

Reply:
`11` is around since long time ago, so I don't mind to keep it, but I don't like it because it doesn't have any meaning / logic. I think it's also a bit low when scales are large and so more ticks could be displayed *by default*.